app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoRepo.kt

```diff
@@ -40,9 +40,9 @@ open class PocketVideoRepo(
     }
 
     private val _feedState = MutableLiveData<FeedState>().apply {
-        // _feedState.value should always be initialized at the top of init,
-        // because we treat it as non-null
-        value = buildConfigDerivables.initialPocketRepoState
+        // setValue must be called from the main thread, and this repo is instantiated
+        // off of the main thread in Espresso tests, so postValue is used instead
```

Comment:
No need to add a comment: why is the repo initialized off the main thread? I don't actually know how threading works in tests (e.g. is the main thread different from the thread running the junit code?).

fwiw, I don't love that the testing harness is impacting our production code: an immediate set is much simpler to handle than the post, e.g. the posted value may not be received by the calling application if another value is posted quickly: is that a problem?. I'm not sure what a solution would be here though without better understanding the test threading.

Reply:
Test code can be run on the main thread using `ActivityTestRule#runOnUiThread`, but by default it runs on a thread created by our `AndroidJunitTestRunner`.  In this case, we need to create the object before `Application` is initialized, so the `ActivityTestRule` thread isn't yet available.

I agree with your discomfort here, and will give it more thought.  I don't think it should block the merge, but hopefully we can come up with a better solution in the future.